cloudant-client/src/main/java/com/cloudant/client/org/lightcouch/ReplicatorDocument.java

```diff
@@ -77,16 +77,65 @@ public String getTarget() {
     }
 
     private String getEndpointUrl(JsonElement element) {
+        if (element == null) {
+            return null;
+        }
         JsonPrimitive urlString = null;
         if (element.isJsonPrimitive()) {
             urlString = element.getAsJsonPrimitive();
         } else {
             JsonObject replicatorEndpointObject = element.getAsJsonObject();
             urlString = replicatorEndpointObject.getAsJsonPrimitive("url");
         }
+        if (urlString == null) {
+            return null;
+        }
         return urlString.getAsString();
     }
 
+    private JsonElement getDestination(JsonElement oldDestination, String newDestination) {
+        if (oldDestination == null || oldDestination.isJsonPrimitive()) {
+            return new JsonPrimitive(newDestination);
+        }
+        JsonObject json = oldDestination.getAsJsonObject();
+        json.addProperty("url", newDestination);
```

Comment:
Should this remove any old `url` property first?
I am reasonably sure that this will work as is (at least currently), but there is a small risk here because of two factors:
i) As per [RFC8259 Section 4](https://tools.ietf.org/html/rfc8259#section-4) `The names within an object SHOULD be unique.`
ii) As per the[javadoc for JsonObject](https://static.javadoc.io/com.google.code.gson/gson/2.8.2/com/google/gson/JsonObject.html#addProperty-java.lang.String-java.lang.String-) `The member elements of this object are maintained in order they were added.`

So as it currently stands your new `url` would, I believe, supersede any previous `url` property on the object when it is serialized. The risk is that a change in GSON behaviour or a swap to another JSON provider could make this break.

Reply:
Added the remove.